tests/integration/test_core.py

```diff
@@ -540,3 +545,111 @@ async def test_openid_plugin(
         assert (
             "administrator" in wordpress_client.list_roles()
         ), "An launchpad OpenID account should be associated with the WordPress admin user"
+
+
+async def test_prometheus_integration(
+    ops_test: pytest_operator.plugin.OpsTest,
+    application_name: str,
+    unit_ip_list: list[str],
+):
+    """
+    arrange: after WordPress charm has been deployed and relations established.
+    act: prometheus charm joins relation
+    assert: prometheus metrics endpoint for prometheus is active and prometheus has active scrape
+        targets.
+    """
+    assert ops_test.model
+    prometheus: Application = await ops_test.model.deploy(
+        "prometheus-k8s", channel="stable", trust=True
+    )
+
+    await ops_test.model.relate(application_name, prometheus.name)
+    await ops_test.model.wait_for_idle(apps=[application_name, prometheus.name], status="active")
+
+    for unit_ip in unit_ip_list:
+        requests.get(
+            f"http://{unit_ip}:{APACHE_PROMETHEUS_SCRAPE_PORT}", timeout=10
+        ).raise_for_status()
+    status: FullStatus = await ops_test.model.get_status(filters=[prometheus.name])
+    for unit in status.applications[prometheus.name].units.values():
+        query_targets = requests.get(
+            f"http://{unit.address}:9090/api/v1/targets", timeout=10
+        ).json()
+        assert len(query_targets["data"]["activeTargets"])
+
+
+async def test_loki_integration(
+    ops_test: pytest_operator.plugin.OpsTest,
+    application_name: str,
+    kube_core_client: kubernetes.client.CoreV1Api,
+):
+    """
+    arrange: after WordPress charm has been deployed and relations established.
+    act: loki charm joins relation
+    assert: loki joins relation successfully, logs are being output to container and to files for
+        loki to scrape.
+    """
+    assert ops_test.model
```

Comment:
An alternative to having this on every test is to create a fixture that exposes the `ops_test.model` so that the assert only needs to be done once